widgets/common/src/filters/RulesForm.tsx

```diff
@@ -1,49 +1,59 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import type { FunctionComponent } from 'react';
 
 import RuleRow from './RuleRow';
 import AddRuleButton from './AddRuleButton';
 import type { FilterRule, FilterRuleRow } from './types';
 import { FilterRuleType, FilterRuleOperators, FilterRuleRowType } from './types';
+import { isAnyOperator } from './common';
+
+function hasError({ type, key, operator, values }: FilterRule) {
+    const noValuesDefined = !values?.length;
+
+    if (type === FilterRuleType.Label) {
+        const noKeyDefined = !key;
+        return noKeyDefined || (isAnyOperator(operator) && noValuesDefined);
+    }
+
+    return noValuesDefined;
+}
 
 function getNewRow(): FilterRuleRow {
     const { uuid } = Stage.Utils;
+    const emptyRule = {
```

Comment:
Optional:

There's a missing type annotation here

```suggestion
    const emptyRule: FilterRule = {
```

Reply:
Thx, added type annotation.